### PR TITLE
postinst:(su) run the Workstation App as mortal user

### PR DESCRIPTION
**Pull request by afiune**
## Description
Update the `postinst` to run the Workstation App as a mortal user instead of `root`.

Signed-off-by: Salim Afiune <afiune@chef.io>

## Related Issue

Depends on https://github.com/chef/chef-workstation-app/pull/160

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by afiune** [comment]
Launched an [ad hoc build](https://buildkite.com/chef/chef-chef-workstation-master-omnibus-adhoc/builds/209#c0e077ca-8e19-4bb3-a5e2-63eb58406fc0)

**Comment by afiune** [comment]
Uhg, this might be wrong and the right environment variable to use is `SUDO_USER=afiune`

tenor-73325880

**Comment by afiune** [comment]
I had a problem running the ad hoc build, the error message is: 
```
Feb 11 13:57:12 Salims-MBP installd[1191]: ./postinstall: /opt/chef-workstation/bin/chef_workstation_app_launcher:set:20: no such option: u
```

I think I need to fix the `chef_workstation_app_launcher` script.

**Comment by afiune** [comment]
New ad hoc build [here](https://buildkite.com/chef/chef-chef-workstation-master-omnibus-adhoc/builds/210#03c57b78-f172-43ed-ade4-6c9ee84dad87)!

File: omnibus/package-scripts/chef-workstation/postinst [hunk 1]
```diff
@@ -58,7 +58,11 @@ if is_darwin; then
 
   # Restart the app if it was running.
   echo "Restarting Chef Workstation App..."
-  $INSTALLER_DIR/bin/$app_launcher load
+  echo "USER: $USER"
```
Comment (marcparadise) [inline omnibus/package-scripts/chef-workstation/postinst]: Could we remove the echos? The output is not actionable or informative to the operator .

Comment (jonsmorrow) [reply omnibus/package-scripts/chef-workstation/postinst]: Agree.

File: omnibus/package-scripts/chef-workstation/postinst [hunk 2]
```diff
@@ -58,7 +58,11 @@ if is_darwin; then
...
+  echo "USERNAME: $USERNAME"
+  echo "SUDO_USER: $SUDO_USER"
+  echo "HOME: $HOME"
```
Comment (jonsmorrow) [inline omnibus/package-scripts/chef-workstation/postinst]: It doesn't appear $HOME is used anywhere?

File: omnibus/package-scripts/chef-workstation/postinst (hunk 1 above)
Comment (afiune) [reply omnibus/package-scripts/chef-workstation/postinst]: 100%!! I need to:

1) wait for the Chef Workstation App to bump its version
2) rebase this branch
3) remove the echos
4) merge!

This build is just for debugging as the [commit message says](https://github.com/chef/chef-workstation/pull/938/commits/cf969b278c776b26dcbde99acb1490423feeeef0) it. I will remove that commit in a bit. thanks for the prompt review

**Comment by afiune** [comment]
https://github.com/chef/chef-workstation/pull/940 Merged the Tray App is now version `0.1.67` with the fix https://github.com/chef/chef-workstation-app/pull/160

File: omnibus/package-scripts/chef-workstation/postinst (hunk 1 above)
Comment (afiune) [reply omnibus/package-scripts/chef-workstation/postinst]: @jonsmorrow @marcparadise could you review again now? I have removed the echos and rebased this branch! 💯

**Comment by afiune** [comment]
Last ad hoc build! Let us wait for 2 hours for the build and then let us verify the functionality, if everything looks good then l will merge this PR.

https://buildkite.com/chef/chef-chef-workstation-master-omnibus-adhoc/builds/211#e49c6e34-ba27-4cdd-94a7-8ca92ea880a6

**Comment by afiune** [comment]
@marcparadise I think that is how omnibus runs the `postinst` script, isn't it? 🤔

**Comment by marcparadise** [comment]
@afiune postinst is run by the platform package manager (rpm, dpkg); omnibus is responsible for putting it in the right place in the package itself.

**Comment by afiune** [comment]
Good that I tested!

We need to use `USER` and not `SUDO_USER`

```
Feb 11 16:18:43 Salims-MacBook-Pro installd[1191]: ./postinstall: Restarting Chef Workstation App...
Feb 11 16:18:43 Salims-MacBook-Pro installd[1191]: ./postinstall: USER: afiune
Feb 11 16:18:43 Salims-MacBook-Pro installd[1191]: ./postinstall: USERNAME: 
Feb 11 16:18:43 Salims-MacBook-Pro installd[1191]: ./postinstall: SUDO_USER: 
Feb 11 16:18:43 Salims-MacBook-Pro installd[1191]: ./postinstall: HOME: /Users/afiune
Feb 11 16:18:43 Salims-MacBook-Pro installd[1191]: ./postinstall: su: unknown login: 
```

I will update this PR! 🥇

**Comment by afiune** [comment]
This is the good one!! This [ad hoc build](https://buildkite.com/chef/chef-chef-workstation-master-omnibus-adhoc/builds/212#74d2eaf4-5167-445a-9164-e914e28fab56) should work!

**Pull request merged**